frontend/generateCard.js

```diff
@@ -1,65 +1,74 @@
+        if (!response.ok) {
+            throw new Error(`HTTP error! status: ${response.status}`);
+        }
+        const json = await response.json();
+        cardsData = json; //Full api data
 
-// Function to create a card element with a title and description
-function makeCard(title, description) {
-    // Create a new div element for the card
+        // Taking the keyword titles and descriptions
+        cardsData.forEach(card => {
+            keywordData.push(card.keyword);
+            descriptionData.push(card.description);
+        });
+    }
+    catch (error) {
+        console.error('Error fetching cards data:', error.message);
+    }
+}
+
+// Creates a card element with a title, description and a line break between them
+function createCard(title, description) {
+    // Create a new div element for the card and adds 'card' and 'visible' class
     const card = document.createElement('div');
-    card.classList.add('card');
-    card.classList.add('visible');
+    card.classList.add('card', 'visible');
 
-    // Create divs for title, separator, description
-    const cardTitle = document.createElement('div');
-    cardTitle.classList.add('cardTitle');
+    const cardTitle = document.createElement('div'); // Create div for title
+    cardTitle.classList.add('cardTitle'); // Give it the class 'cardTitle'
     cardTitle.textContent = title;
 
-    const cardLineBreak = document.createElement('div');
-    cardLineBreak.classList.add('cardLineBreak');
+    const cardLineBreak = document.createElement('div'); // Create div for the linebreak
+    cardLineBreak.classList.add('cardLineBreak'); // Give it the class 'cardLineBreak'
 
-    const cardDescription = document.createElement('div');
-    cardDescription.classList.add('cardDescription');
+    const cardDescription = document.createElement('div'); // Create div for description
+    cardDescription.classList.add('cardDescription'); // Give it the class 'cardDescription'
     cardDescription.textContent = description;
 
     // Append title, line break, and description to the card
     card.appendChild(cardTitle);
     card.appendChild(cardLineBreak);
     card.appendChild(cardDescription);
-    // Return the constructed card element
-    return card;
-}
 
-// Function to generate and append cards based on titles and descriptions arrays
-function generateCards() {
-    for (let i = 0; i < titles.length; i++) {
-        const cards = makeCard(titles[i], descriptions[i]);
-        cardHolder.appendChild(cards);
+    return card; // Return the constructed card element
+}
+// Renders all the cards on the page
+async function renderCards() {
+    try {
+        for (let i = 0; i < keywordData.length; i++) {
+            try {
+                const card = createCard(keywordData[i], descriptionData[i]);
+                cardHolder.appendChild(card);
+            } catch (innerError) {
+                console.log('Error rendering card:', innerError.message);
```

Comment:
If we replace the outer for loop with a foreach we can print the attributes in the log of the card which failed to make it easier to debug.